### PR TITLE
mgis: add new variant for static libraries

### PR DESCRIPTION
**Pull request by Briffou**
@thelfer  I'm not sure this is the right way to add this in spack (I saw a lot of threads about static vs shared), feel free to close eventually

### CODE REVIEW

**Review by thelfer — APPROVED** [review]
fine for me

**Comment by thelfer** [comment]
@Briffou To be honest, normally we should have used the standard `cmake` mechanism, i.e. use the  BUILD_SHARED_LIBS variable to select if we want to build shared libraries or a static libraries.

In `MGIS` (in `TFEL` to be more precise), I made the choice long ago (more than 10 years) to always compile shared libraries and to optionaly compile static libraries using the `enable-static` flag. Maybe I should do something more standard...

However, you are the first one to request a static variant of `MGIS`, so let me know if the current solution is ok for you or if you.

**Comment by Briffou** [comment]
@thelfer , sure I understand. Current solution if also ok for me.

File: var/spack/repos/builtin/packages/mgis/package.py [hunk 1]
```diff
@@ -104,4 +105,7 @@ def cmake_args(self):
             # adding path to boost
             args.append("-DBOOST_ROOT={0}".format(self.spec["boost"].prefix))
 
+        if "+static" in self.spec:
+            args.append("-Denable-static=ON")
+
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/mgis/package.py:110]: FYI. You could use the `self.define_from_variant()` helper here if you want.  See https://spack.readthedocs.io/en/latest/build_systems/cmakepackage.html#adding-flags-to-cmake.

**Review by tldahlgren — APPROVED** [review]
LGTM.  Let me know if you want to use the helper or keep the code as-is.

File: var/spack/repos/builtin/packages/mgis/package.py (hunk 1 above)
Comment (Briffou) [reply var/spack/repos/builtin/packages/mgis/package.py:110]: Didn't know, thanks @tldahlgren. 

@thelfer I let you make the final decision 😉

Comment (thelfer) [reply var/spack/repos/builtin/packages/mgis/package.py:110]: Ok for me

**Pull request merged**